store/postgres/secret_repository.go

```diff
@@ -86,56 +85,81 @@ func (p Secret) ToSpec(hash models.ApplicationKey) (models.ProjectSecretItem, er
 	}, nil
 }
 
+func (p Secret) ToSecretItemInfo(hash models.ApplicationKey) (models.SecretItemInfo, error) {
+	// decode base64
+	encrypted, err := base64.StdEncoding.DecodeString(p.Value)
+	if err != nil {
+		return models.SecretItemInfo{}, err
+	}
+
+	digest := cryptopasta.Hash("user defined secrets", encrypted)
+	base64encoded := base64.StdEncoding.EncodeToString(digest)
+
+	// Todo: Move to Secret type
+	secretType := models.SecretTypeSystemDefined
+	if p.Type == models.SecretTypeUserDefined.String() {
+		secretType = models.SecretTypeUserDefined
+	}
+
+	return models.SecretItemInfo{
+		ID:        p.ID,
+		Name:      p.Name,
+		Digest:    base64encoded,
+		Type:      secretType,
+		Namespace: p.Namespace.Name,
+		UpdatedAt: p.UpdatedAt,
+	}, nil
+}
+
 type secretRepository struct {
-	db        *gorm.DB
-	project   models.ProjectSpec
-	namespace models.NamespaceSpec
+	db      *gorm.DB
+	project models.ProjectSpec
 
 	hash models.ApplicationKey
 }
 
-func (repo *secretRepository) Insert(ctx context.Context, resource models.ProjectSecretItem) error {
-	p, err := Secret{}.FromSpec(resource, repo.project, repo.namespace, repo.hash)
+func (repo *secretRepository) Insert(ctx context.Context, namespace models.NamespaceSpec, resource models.ProjectSecretItem) error {
+	p, err := Secret{}.FromSpec(resource, repo.project, namespace, repo.hash)
 	if err != nil {
 		return err
 	}
-	if len(p.Name) == 0 {
+	if len(p.Name) == 0 { // TODO: already checked in service
 		return errors.New("name cannot be empty")
 	}
 	return repo.db.WithContext(ctx).Save(&p).Error
 }
 
-func (repo *secretRepository) Save(ctx context.Context, spec models.ProjectSecretItem) error {
+func (repo *secretRepository) Save(ctx context.Context, namespace models.NamespaceSpec, spec models.ProjectSecretItem) error {
 	_, err := repo.GetByName(ctx, spec.Name)
 	if errors.Is(err, store.ErrResourceNotFound) {
-		return repo.Insert(ctx, spec)
+		return repo.Insert(ctx, namespace, spec)
 	} else if err != nil {
 		return errors.Wrap(err, "unable to find secret by name")
 	}
-	return errors.New("secret already exist")
+	return store.ErrResourceExists
```

Comment:
we can log the secret with the name already exists, as the error that we are returning is generic error